test/Test_Elaboration.re

```diff
@@ -7,90 +7,105 @@ let dhexp_typ = testable(Fmt.using(Exp.show, Fmt.string), DHExp.fast_equal);
 
 let ids = List.init(12, _ => Id.mk());
 let id_at = x => x |> List.nth(ids);
+
 let mk_map = Statics.mk(CoreSettings.on, Builtins.ctx_init);
 let dhexp_of_uexp = u => Elaborator.elaborate(mk_map(u), u) |> fst;
 let alco_check = dhexp_typ |> Alcotest.check;
 
-let u1: Exp.t = {ids: [id_at(0)], term: Int(8), copied: false};
-let single_integer = () =>
-  alco_check("Integer literal 8", u1, dhexp_of_uexp(u1));
-
-let u2: Exp.t = {ids: [id_at(0)], term: EmptyHole, copied: false};
-let empty_hole = () => alco_check("Empty hole", u2, dhexp_of_uexp(u2));
-
-let u3: Exp.t = {
-  ids: [id_at(0)],
-  term: Parens({ids: [id_at(1)], term: Var("y"), copied: false}),
-  copied: false,
-};
-
-let free_var = () => alco_check("free variable", u3, dhexp_of_uexp(u3));
+let alco_check_menhir = (name: string, dhexp: string, uexp: Term.Exp.t) =>
```

Comment:
Also now that dhexp and uexp are merged how do you feel about this:

```reason
let alco_check_menhir = (name: string, dhexp: string, uexp: string) =>
  alco_check(
    name,
    Haz3lmenhir.Conversion.Exp.of_menhir_ast(
      Haz3lmenhir.Interface.parse_program(dhexp),
    ),
    dhexp_of_uexp(
      Haz3lmenhir.Conversion.Exp.of_menhir_ast(
        Haz3lmenhir.Interface.parse_program(uexp),
      ),
    ),
  );
  ```
  
  That way you can write the tests like
  ```reason
  alco_check_menhir2(
    "Let expression for a function which is not recursive (menhir)",
    "let f =
    fun x ->
        1 + x<Unknown Internal => Int>
    f
    in
    55",
    "let f =
          fun x ->
              1 + x
          f
          in
      55",
  );
  ```

Reply:
As far as changing the alco_check_menhir to allow textual uexps, I would be ok with that but I do have a slight concern that if we do that the elaborator tests really become Hazel parser + elaborator tests and I think it might be a bit better to keep the parser/elaborator tests separate - what do you think about that?

Reply:
> Can we rename `dhexp` to `expected` and `uexp` to `actual` so it's clear that the assertion is the dhexp and the thing we're testing is the uexp.

Yeah totally - especially after the uexp/dhexp merge it makes more sense to do actual/expected.

Reply:
> As far as changing the alco_check_menhir to allow textual uexps, I would be ok with that but I do have a slight concern that if we do that the elaborator tests really become Hazel parser + elaborator tests and I think it might be a bit better to keep the parser/elaborator tests separate - what do you think about that?

I agree that they'll be parser + elaborator tests. But I think that's true if we do it just for the actual or for the actual + expected. If we wanted to decouple the tests we'd probably need to remove the parser from both which seems _too_ expensive. If we have good enough parser tests hopefully we can build up some confidence on the parsing side and trust these test failures are mostly around the elaboration. In the mean time we can keep the old elaboration tests.